### PR TITLE
Splitting 3 database3.py classes into 3 files

### PR DESCRIPTION
**Pull request by john-science**
## Description

The old `database3.py` file had 3 classes and several functions in it.  But a better standard in Python is "1 class = 1 file" for big, meaty classes like this.

Also, I was not sure it was obvious to people what all was in this file.  

Since the default GitHub diff for this PR is so large, I tried to keep my commits atomic, so you could see what all was being done more easily.  So, I would suggest looking at the commits, and not the diff. Though, to help, here is a brief synopsis of the changes in this PR:

* `db/database3.py:Database3` → no change
* `db/database3.py:DatabaseInterface`→ `db/databaseInterface.py:DatabaseInterface`
* `db/database3.py:Layout` → `db/layout.py:Layout`
* `bookkeeping/tests/test_databaseInterface.py` → `bookkeeping/db/tests/test_databaseInterface.py`

I also updated the major docstrings for `db/database3.py`, db/__init__.py`, and `Layout` to be up-to-date.

---

## Checklist

- [X] This PR has only one purpose or idea.
- [X] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

**Comment by keckler** [comment]
Just a note to any users, this changes where you need to import the `databaseInterface` from, if you import it directly.

**Comment by keckler** [comment]
Yeah, probably most people pull up that interface via `getInterface(name="database")`, because like you said it is always generated in a run.

I think its a good change. Just highlighting the API change.

**Review by opotowsky — APPROVED** [review]
Wow, great work! I found one downstream import of `DatabaseInterface`, but that doesn't block this in any way. 

I didn't look hard at the tests or much of the code since it was just being moved around, but I did follow your commits as suggested so I did see the changes you made. 

I think it looks really good.

File: doc/release/0.2.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ Release Date: TBD
 
 What's new in ARMI
 ------------------
+#. Split 3 classes in ``da5abase3.py`` into 3 files (`PR#955 <https://github.com/terrapower/armi/pull/923>`_)
```
[the remark this answers is not in the document]
Comment (john-science) [reply doc/release/0.2.rst]: Fixed

**Comment by john-science** [comment]
Thanks for the save @opotowsky !

**Pull request merged**